mkosi/installer/apk.py

```diff
@@ -0,0 +1,121 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+from collections.abc import Iterable, Sequence
+from pathlib import Path
+
+from mkosi.config import Config
+from mkosi.context import Context
+from mkosi.installer import PackageManager
+from mkosi.run import run, CompletedProcess
+from mkosi.util import _FILE, PathString
+
+
+class Apk(PackageManager):
+    @classmethod
+    def executable(cls, config: Config) -> str:
+        return "apk"
+
+    @classmethod
+    def subdir(cls, config: Config) -> Path:
+        return Path("apk")
+
+    @classmethod
+    def package_subdirs(cls, cache: Path) -> list[tuple[Path, Path]]:
+        return [(Path("apk"), Path("apk"))]
+
+    @classmethod
+    def scripts(cls, context: Context) -> dict[str, list[PathString]]:
+        return {
+            "apk": cls.apivfs_script_cmd(context) + cls.cmd(context),
+            "mkosi-install":   ["apk", "--update-cache", "add"],
+            "mkosi-upgrade":   ["apk", "--update-cache", "upgrade"],
+            "mkosi-remove":    ["apk", "--remove", "del"],
+            "mkosi-reinstall": ["apk", "--update-cache", "fix"],
+        }  # fmt: skip
+
+    @classmethod
+    def mounts(cls, context: Context) -> list[PathString]:
+        mounts: list[PathString] = [
+            *super().mounts(context),
+        ]
+
+        mounts += ["--ro-bind", "/etc/apk/keys", "/etc/apk/keys"]
+
+        cache_dir = context.config.package_cache_dir_or_default() / "apk"
+        cache_dir.mkdir(parents=True, exist_ok=True)
+        mounts += ["--bind", cache_dir, "/var/cache/apk"]
+
+        return mounts
+
+    @classmethod
+    def setup(cls, context: Context, repositories: Iterable[str]) -> None:
+        # Create cache symlink to enable automatic caching
+        cache_link = context.root / "etc/apk/cache"
+        if not cache_link.exists():
+            cache_link.parent.mkdir(exist_ok=True, parents=True)
+            cache_link.symlink_to("/var/cache/apk")
+
+        config = context.root / "etc/apk/repositories"
+        if config.exists():
+            print("apk setup() repo file exists!")
+            return
```

Comment:
I think we'll just want to use `--repositories-file` instead assuming it's not interpreted relative to the root directory.